job_executor/model/datastore.py

```diff
@@ -131,55 +154,123 @@ def change_data(self, dataset_name: str, description: str):
             draft_metadata.dict(by_alias=True), dataset_name, 'DRAFT'
         )
         local_storage.move_working_dir_parquet_to_datastore(dataset_name)
+        job_service.update_job_status(job_id, 'completed')
+        self._log(job_id, 'completed')
 
-    def remove(self, dataset_name: str, description: str):
+    def remove(self, job_id: str, dataset_name: str, description: str):
         """
         Remove a released dataset that has been released in
         a previous version from future versions of the datastore.
         """
+        self._log(job_id, 'initiated')
+        job_service.update_job_status(job_id, 'initiated')
         dataset_release_status = self._get_release_status(dataset_name)
-        if dataset_release_status != 'RELEASED':
-            raise VersioningException(
-                'Can\'t remove dataset with release status '
-                f'{dataset_release_status}'
+        dataset_is_draft = (
+            self.draft_version.get_dataset_release_status(dataset_name)
+            is not None
+        )
+        dataset_operation = (
+            self.draft_version.get_dataset_operation(dataset_name)
+        )
+        if dataset_is_draft and dataset_operation == 'REMOVE':
+            self.metadata_all_draft.remove(dataset_name)
+            log_message = 'Dataset already in draft with operation REMOVE.'
+            self._log(job_id, log_message)
+            job_service.update_job_status(job_id, 'completed', log_message)
```

Comment:
Excuse the long answer, it's a good question 😅 

If a REMOVE of dataset_x job is requested the two things we want to make sure of is:
    * There is no metadata for dataset_x in metadata_all__DRAFT
    * There is a REMOVE dataset_x in datastructureUpdates
 If we consider both of these to be the case, the job is considered completed.
 
 If the job is interrupted after removing it from metadata_all and adding the appropriate object to datastructureUpdates, the re-queueing of the job simply makes sure this is still true with this conditional and marks the job done. The state the job requested is already validated to be so.
The other case (apart from an interrupt) where this could happen is if a mishap in the frontend causes a REMOVE job to be accepted when a identical REMOVE job is already completed. In that case we won't get a 'failed' for the job, but a 'completed' with the additional log message: 'dataset_x already in draft with operation REMOVE.'

I can't think of other edge cases where we enter this conditional where it would rob us of information due to a lack of "failed". I might have missed something here so feel free to hit me up with a message/call if I'm missing a case 👍🏻

Reply:
Sounds good!